### PR TITLE
Do not truncate body if `request_bodies` is `"always"`

### PR DESCRIPTION
**Pull request by sentrivana**
We've received quite some feedback regarding our automatic request body trimming making it harder to debug issues. This PR changes the behavior of our serializer so that we don't apply our `MAX_DATABAG_BREADTH` and `MAX_DATABAG_DEPTH` limits if the SDK has been initialized with `request_bodies="always"`.

Fixes https://github.com/getsentry/sentry-python/issues/1041

### CODE REVIEW

File: sentry_sdk/serializer.py [hunk 1]
```diff
@@ -253,10 +277,18 @@ def _serialize_node_impl(
         if is_databag is None:
             is_databag = _is_databag()
 
-        if is_databag and remaining_depth is None:
-            remaining_depth = MAX_DATABAG_DEPTH
-        if is_databag and remaining_breadth is None:
-            remaining_breadth = MAX_DATABAG_BREADTH
+        if is_request_body is None:
+            is_request_body = _is_request_body()
```
Comment (sl0thentr0py) [inline sentry_sdk/serializer.py:281]: is this needed? not sure if it ever is true if it's already not true above?

File: sentry_sdk/serializer.py [hunk 2]
```diff
@@ -118,6 +120,10 @@ def serialize(event, **kwargs):
     path = []  # type: List[Segment]
     meta_stack = []  # type: List[Dict[str, Any]]
 
+    do_not_trim_request_bodies = (
```
Comment (antonpirker) [inline sentry_sdk/serializer.py:123]: Nitpicking: positive vars are easier to comprehend when used in conditions. Something like `should_trim_request_bodies`. (nitpicking stuff you can always decide to not change, and just go with your solution if you like it better)

File: sentry_sdk/serializer.py [hunk 3]
```diff
@@ -182,10 +188,11 @@ def _is_databag():
             if rv in (True, None):
                 return rv
 
-            p0 = path[0]
-            if p0 == "request" and path[1] == "data":
-                return True
+            is_request_body = _is_request_body()
+            if is_request_body in (True, None):
```
Comment (antonpirker) [inline sentry_sdk/serializer.py:192]: I do not get why there is now a "None" case? Can you explain this to me"

**Review by sl0thentr0py — APPROVED** [review]
lgtm! please test a bit with an app with errors and tranasctions just in case because the serializer is fairly magical and is a core code path

File: sentry_sdk/serializer.py (hunk 2 above)
Comment (sentrivana) [reply sentry_sdk/serializer.py:123]: Makes sense, will change. Didn't like the name in the first place.

File: sentry_sdk/serializer.py (hunk 3 above)
Comment (sentrivana) [reply sentry_sdk/serializer.py:192]: `None` basically means we're currently recursively looking at a data level where there is no `['request']['data']`, but it's still possible that we'll encounter it somewhere further down the tree (as opposed to `False` which means we're somewhere where one can't expect a request body to appear anymore). The behavior basically copies what we do [here](https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/serializer.py#L148-L152) and [here](https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/serializer.py#L177-L178).

File: sentry_sdk/serializer.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/serializer.py:279]: Oh yeah, thanks for spotting that

File: sentry_sdk/serializer.py (hunk 2 above)
Comment (sentrivana) [reply sentry_sdk/serializer.py]: Changed to `keep_request_bodies`

File: sentry_sdk/serializer.py (hunk 3 above)
Comment (antonpirker) [reply sentry_sdk/serializer.py:190]: oh, now I get it. Thanks for explaining!

**Pull request merged**